packages/rum-core/src/domain/rumEventsCollection/view/viewCollection.ts

```diff
@@ -1,63 +1,65 @@
 import { Duration, isEmptyObject, mapValues, ServerDuration, toServerDuration } from '@datadog/browser-core'
 import { RawRumViewEvent, RumEventType } from '../../../rawRumEvent.types'
 import { LifeCycle, LifeCycleEventType } from '../../lifeCycle'
+import { ForegroundContexts } from '../../foregroundContexts'
 import { trackViews, ViewEvent } from './trackViews'
 
-export function startViewCollection(lifeCycle: LifeCycle, location: Location) {
+export function startViewCollection(lifeCycle: LifeCycle, location: Location, foregroundContexts: ForegroundContexts) {
   lifeCycle.subscribe(LifeCycleEventType.VIEW_UPDATED, (view) =>
     lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, processViewUpdate(view))
   )
 
-  return trackViews(location, lifeCycle)
-}
-
-function processViewUpdate(view: ViewEvent) {
-  const viewEvent: RawRumViewEvent = {
-    _dd: {
-      document_version: view.documentVersion,
-    },
-    date: view.startClocks.timeStamp,
-    type: RumEventType.VIEW,
-    view: {
-      action: {
-        count: view.eventCounts.userActionCount,
-      },
-      cumulative_layout_shift: view.cumulativeLayoutShift,
-      dom_complete: toServerDuration(view.timings.domComplete),
-      dom_content_loaded: toServerDuration(view.timings.domContentLoaded),
-      dom_interactive: toServerDuration(view.timings.domInteractive),
-      error: {
-        count: view.eventCounts.errorCount,
+  function processViewUpdate(view: ViewEvent) {
```

Comment:
any reason to move processViewUpdate inside startViewCollection?

passing foregroundContext as parameter instead could ease the review

Reply:
nit: I'd be in favor to do it, it would also avoid to mess the history for those lines.